lib/rmt/mirror.rb

```diff
@@ -120,7 +120,11 @@ def mirror_metadata(repository_dir, repository_url, temp_metadata_dir)
 
     metadata_files
   rescue StandardError => e
-    raise RMT::Mirror::Exception.new(_('Error while mirroring metadata: %{error}') % { error: e.message })
+    unless do_not_raise && e.message.include?('GPG signature verification failed')
```

Comment:
This looks prone to errors, to be honest. What kind of exception is raised whenever a GPG error occurs? Maybe we can put this logic in a separate `rescue` for the specific exception...